dev_requirements.txt

```diff
@@ -15,4 +15,5 @@ retworkx>=0.10.2
 setuptools>=62.3.2
 scipy>=1.6
 thewalrus>=0.19.0
+plotly>=4.5.0

```

Comment:
I wonder if we should make this an optional requirement? i.e., not adding it to the `dev_requirements` file. For it to actually be usable it also needs jupyter notebook. 🤔

If the latter option makes sense, we can add a warning message if the user tries to use plotly and fallback to matplotlib. The warning message can contain the instructions to install plotly (as well as the documentation).

Reply:
Afaik, plotly is usable in other environments outside of notebooks, but it might open a browser tab to display the figures (instead of showing them inline). It seems to me that just adding plotly to `dev_requirements` is the simplest solution that's also user-friendly. Unless plotly is not supported by some machines, I'd propose keeping it there.

Reply:
FYI - plotly provides a couple of renderers https://plotly.com/python/renderers/. Depending on the IDE, one renderer could be preferred. For many common IDEs/environments plotly sets the best renderer automatically but not for all. I added a note in the docstring so users are aware of this.